tools/cli/adminKafkaCommands.go

```diff
@@ -594,8 +636,23 @@ func AdminRereplicate(c *cli.Context) {
 	target := getRequiredOption(c, FlagTargetCluster)
 	targets := []string{target}
 
-	producer := newKafkaProducer(c)
 	session := connectToCassandra(c)
+	adminClient := cFactory.ServerAdminClient(c)
+	var startVersion *int64
+	var producer messaging.Producer
+	if c.IsSet(FlagStartEventVersion) {
+		version := c.Int64(FlagStartEventVersion)
+		startVersion = &version
```

Comment:
`startVersion = common.Int64Ptr(c.Int64(FlagStartEventVersion))`? I think it will be easier to read.

Took me sometime to understand why you created another local `version` variable here.

Reply:
Yes. Updated